code/modules/food_and_drinks/food/snacks_meat.dm

```diff
@@ -179,6 +179,20 @@
 		visible_message("<span class='notice'>[src] fails to expand!</span>")
 	qdel(src)
 
+/obj/item/reagent_containers/food/snacks/monkeycube/suicide_act(mob/user)
+	var/mob/living/M = user
+	user.visible_message("<span class='suicide'>[M] is putting [src] in [M.p_their()] mouth! It looks like [M.p_theyre()] trying to commit suicide!</span>")
+	sleep(10)
+	playsound(M.loc,'sound/items/eatfood.ogg', rand(10,50), TRUE)
+	if(HAS_TRAIT(M, TRAIT_NOHUNGER))
+		to_chat(user, "<span class='warning'>Your body won't activate [src]...</span>")
+		return SHAME
+	sleep(10)
+	Expand()
+	M.visible_message("<span class='danger'>[M]'s torso bursts open as a primate emerges!</span>")
+	M.gib(null, TRUE, null, TRUE)
+	return MANUAL_SUICIDE
```

Comment:
This is shit. It doesn't even remove the cube from their hands. You could just throw it away and it will emerge in another room while you still get gibbed. Shit flavour implementation.
Also needs QDELETED checks for src and M could runtime otherwise.

Reply:
other delayed suicide acts didn't have checks for the item still being in the user's hands, so i didn't consider (re)moving it the item. not sure where to put the qdeleted check for src, as Expand() qdeletes it anyway (and, this isn't an antagonistic question, but one so i can better understand when and where i need to put qdeleted checks: why do i need to check if M has been qdeleted? it doesn't happen in every proc after every call to an atom, so when is it relevant? just after waits/long computations? what causes the runtime?).

Reply:
qdeleted checks go directly after the sleep. The cube can't expand if it has been deleted while this proc has been sleeping. Same goes the mob.
If they have been deleted while this was on ice, the value in those variables will be null and the proc calls will runtime.